bitnami/clickhouse/templates/certs.yaml

```diff
@@ -88,4 +88,44 @@ spec:
     size: {{ int .Values.tls.autoGenerated.certManager.keySize }}
   duration: {{ .Values.tls.autoGenerated.certManager.duration }}
   renewBefore: {{ .Values.tls.autoGenerated.certManager.renewBefore }}
+{{- if .Values.keeper.enabled }}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ printf "%s-keeper-crt" (include "common.names.fullname" .) }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" . ) | nindent 4 }}
+    app.kubernetes.io/component: keeper
+    app.kubernetes.io/part-of: clickhouse
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" . ) | nindent 4 }}
+  {{- end }}
+spec:
+  secretName: {{ template "clickhouse.tls.keeper.secretName" . }}
+  commonName: {{ printf "%s.%s.svc.%s" (include "clickhouse.keeper.fullname" .) (include "common.names.namespace" .) .Values.clusterDomain }}
+  issuerRef:
+    name: {{ printf "%s-ca-issuer" (include "common.names.fullname" .) }}
+    kind: Issuer
+  subject:
+    organizations:
+      - "ClickHouse"
+  dnsNames:
+    - '*.{{ include "common.names.namespace" . }}'
+    - '*.{{ include "common.names.namespace" . }}.svc'
+    - '*.{{ include "common.names.namespace" . }}.svc.{{ .Values.clusterDomain }}'
+    - '*.{{ include "clickhouse.keeper.fullname" . }}'
+    - '*.{{ include "clickhouse.keeper.fullname" . }}.{{ include "common.names.namespace" . }}'
+    - '*.{{ include "clickhouse.keeper.fullname" . }}.{{ include "common.names.namespace" . }}.svc'
+    - '*.{{ include "clickhouse.keeper.fullname" . }}.{{ include "common.names.namespace" . }}.svc.{{ .Values.clusterDomain }}'
+    - '*.{{ include "clickhouse.keeper.headlessServiceName" . }}'
+    - '*.{{ include "clickhouse.keeper.headlessServiceName" . }}.{{ include "common.names.namespace" . }}'
+    - '*.{{ include "clickhouse.keeper.headlessServiceName" . }}.{{ include "common.names.namespace" . }}.svc'
+    - '*.{{ include "clickhouse.keeper.headlessServiceName" . }}.{{ include "common.names.namespace" . }}.svc.{{ .Values.clusterDomain }}'
```

Comment:
This seems too permissive, too many wildcards and dns names here, in the [autogenerated tls secret](https://github.com/bitnami/charts/blob/12289f5526700d71a4fbb15676b626647d068adb/bitnami/clickhouse/templates/tls-secret.yaml#L28) we are using a different set.  I've just seen the same configuration for the clickhouse certificate. Do you think is worth to be revisited in a different task?

Reply:
Please feel free to open a task to review this internally, this pattern is sth we have on several other charts.